lesson19/src/js/tab.js

```diff
@@ -0,0 +1,173 @@
+    const fragmentTopics = document.createDocumentFragment();
+    for (const topics of newsData){
+        const category = createAttributedElements({
+            tag:"li",
+            valuesByAttributes:{class:"js-topics-list__item topics-list__item"},
+            str:topics.categories
+        });
+        fragmentTopics.appendChild(category);
+    }
+    topicsList.appendChild(fragmentTopics);
+}
+
+const renderImages = (newsData) => {
+    const contentsBox = document.getElementById("js-contents-box");
+    const imageArea = createAttributedElements({
+        tag:"div",
+        valuesByAttributes:{class:"contents-item topics-img"}
+    });
+
+    const fragmentTopicImg = document.createDocumentFragment();
+    for (const topics of newsData){
+        const topicsImg = createAttributedElements({
+            tag:"img",
+            valuesByAttributes:{class:"js-topics-img__item topics-img__item", src:topics.picture
+        }});
+        fragmentTopicImg.appendChild(topicsImg);
+    }
+
+    contentsBox.appendChild(imageArea).appendChild(fragmentTopicImg);
+}
+
+const renderArticlesSection = (newsData) => {
+    const contentsBox = document.getElementById("js-contents-box");
+    const articleArea = createAttributedElements({
+        tag:"div",
+        valuesByAttributes:{class:"contents-item articles"}
+    });
+
+    const fragmentAritcle = document.createDocumentFragment(); 
+    for (const topics of newsData){
+        const articleList = createAttributedElements({
+            tag:"ul",
+            valuesByAttributes:{class:"js-article-list article-list"}
+        });
+        fragmentAritcle.appendChild(articleList)
+        .appendChild(createArticleTitleFlagment(topics));
+    }
+
+    contentsBox.appendChild(articleArea).appendChild(fragmentAritcle); 
+}
+
+const createArticleTitleFlagment = ({articles}) => {
+    const articleTitleFlagment = document.createDocumentFragment(); 
+
+    for(const article of articles ){
+        const articleItem = createAttributedElements({
+            tag:"li",
+            valuesByAttributes:{class:"js-article-list__item article-list__item"}
+        });
+
+        const articleLink = createAttributedElements({
+            tag:"a",
+            valuesByAttributes:{class:"js-article-link"},
+        });
+        
+        const articleTitle = createAttributedElements({
+            tag:"p",
+            valuesByAttributes:{class:"article-title"},
+            str:article.title
+        });
+
+        articleTitleFlagment.appendChild(articleItem).appendChild(articleLink).appendChild(articleTitle);
+
+        isNewAriticle(article.date) && showNewIcon(articleItem);
+        (article.comment.length) && showCommentIconWithNumber(articleItem, article.comment.length);
+    }
+
+    return articleTitleFlagment;
+}
+
+const isNewAriticle = (articleDate) => {
+    const SpecificPeriod = 3;
+    const dateDifferencial = differenceInDays(new Date(), new Date(articleDate));
```

Comment:
IMO
Isn't it `Differential`, not `Differencial`?
Also, I think `differentialDays` would be easier to understand the meaning, but what do you think?
```suggestion
    const differentialDays = differenceInDays(new Date(), new Date(articleDate));
```

Reply:
Yes,Differential is Correct! I mistaked.

Certainly, I think differentialDays is easier to understand.
I fixed!
https://github.com/nakagawamai/js-lesson/pull/21/commits/36001bded87b71591350b5799419b477e51a261d